### ISSUE TITLE
Datasource: devservice is started even if shouldn't

### ISSUE THREAD

**Issue by JiriOndrusek** [quarkusio/quarkus#39189]
### Describe the bug

Imagine a jenkins job which runs test using Oracle datasource. The execution is failing.
Problem happens only if configuration for multiple datasources is present.

Let me explain: 
- the test contains 1 named datasource (name is `oracle`)
- test is executed with overriding parameters for datasource , for both `oracle` and `mysql`
-  `oracle` devservice is disabled by using `-Dquarkus.datasource.oracle.devservices.enabled=false`
-  configuration overrides 'mysql' devservice's container image

### Expected behavior

No devservice is started (`oracle` one is disabled, `mysql` datasource is not present)

### Actual behavior

Devservice based on `mysql` configuration is started - which fails.

### How to Reproduce?

Reproducer:

1 - clone https://github.com/apache/camel-quarkus
2 - run test `integration-test-groups/jdbc/oracle` with parameters `-Dquarkus.datasource.oracle.devservices.enabled=false -Dquarkus.datasource.mysql.devservices.image-name=wrong_name`
3 - test has to fail, because oracle devservice is disabled. 
4 - The reason of the failure should not be the docker pull of image `wrong_name`,  but the failure `quarkus.datasource."oracle".jdbc.url has not been defined`

### Quarkus version or git rev

3.8.0

### Additional information

Is there a workaround?

**Comment by geoand** [comment]
cc @yrodiere

**Comment by yrodiere** [comment]
By passing `-Dquarkus.datasource.mysql.devservices.image-name=wrong_name` you *did* define the `mysql` datasource. All it takes is one configuration property.

By the way, the `mysql` datasource in your example is probably considered an oracle datasource, since you have only the `jdbc-oracle` extension in your classpath. Datasource names have no particular meaning for Quarkus.

I'm not sure what we could do better here... you did define a datasource, and you even set the devservices image name, so it kind of makes sense to try to start that dev service, no?

Maybe if you explained why you set the property `quarkus.datasource.mysql.devservices.image-name`, we could find an alternative solution?

**Comment by JiriOndrusek** [comment]
Hi, the usecase is a jenkins job, which runs several test modules, some of them contains oracle datasource, some of them mysql datasource. From my POV it makes sense to have a bunch of parameters for all of them together in one job..
The issue I see is the fact, that with the correct mysql image-name of devservice, the oracle test fails, because oracle devserv. processor starts the mysql image. Which is not correct.

**Comment by JiriOndrusek** [comment]
Let me rephrase the usecase:
- there is 1 test module, which uses oracle, nysql and other datasources
- jenkins job to run the tests
- for some reasons, the oracle datasource has to use an external db and mysql devservice has to start different image
- parameters to disable oracle (+ set the url) and parameters with the right devservice images for mysql are present
---> should work

unfortunately the oracle devservice starts first and detects mysql image and runs it. (it is executed from datasource processor)

**Comment by yrodiere** [comment]
> (just the fact that oracle dev service starts image defined for a different datasource sounds like a bug)

I didn't check your build, but from what you're doing I'd expect the Oracle dev services starts **for the  datasource named `mysql`**, but **that datasource is an Oracle datasource**.

Forget about the name of the datasource: you could be using `foobar` instead of `mysql`, that doesn't matter.

By setting `quarkus.datasource.mysql.devservices.image-name`, you did define a datasource named `mysql`.

That datasource has no `db-kind` defined, so it'll default to what's in the classpath, and you only have `jdbc-oracle`, so it will be an Oracle datasource.

So that datasource being defined, and dev services being enabled by default, the Oracle dev services will try to start for that datasource. And fail, because the image is wrong.

> Let me rephrase the usecase:
> 
>     * there is 1 test module, which uses oracle, nysql and other datasources
> 
>     * jenkins job to run the tests
> 
>     * for some reasons, the oracle datasource has to use an external db and mysql devservice has to start different image
> 
>     * parameters to disable oracle (+ set the url) and parameters with the right devservice images for mysql are present
>       ---> should work
> 
> 
> unfortunately the oracle devservice starts first and detects mysql image and runs it. (it is executed from datasource processor)

I would suggest indirection.

E.g. instead of passing `-Dquarkus.datasource.mysql.devservices.image-name=something`, you'd pass `-Dmysql.image-name=something`, and in your various `application.properties`, you'd add something like this where relevant (and only where relevant, i.e. not in the Oracle project):

```properties
quarkus.datasource.mysql.devservices.image-name=${mysql.image-name}
```

I think it should work on CI, and should default to Quarkus' default image on local development environments.

**Comment by JiriOndrusek** [comment]
I understand your proposal and it makes sense. 

Thanks for the explanation!

**Comment by yrodiere** [comment]
Please keep us updated; if my suggestion works, I'll close this issue. Otherwise we'll try to find something else :)

**Comment by JiriOndrusek** [comment]
I checked your splution and it should work, thanks for the advice!

For me the major reveal was the fact, that if a module contains only e.g. oracle jdbc dependecy and I add `-Dquarkus.datasource.mysql.devservices.image-name=wrong_name`
I instruct quarkus to run the image by oracle processor.

Since that moment everything makes sense...

**Comment by yrodiere** [comment]
Thanks for the update :)
